app/controllers/api/hosts_controller.rb

```diff
@@ -12,45 +11,28 @@ class HostsController < BaseProviderController
     include Subcollections::Tags
 
     def edit_resource(type, id, data = {})
-      # TODO: drop 'credentials' parameter field when ui-classic hosts is in react
-      credentials = data.delete(CREDENTIALS_ATTR)
       authentications = data.delete(AUTH_ATTR)
+
       raise BadRequestError, "Cannot update non-credentials attributes of host resource" if data.any?
+
       resource_search(id, type).tap do |host|
-        # begin legacy ui-classic
-        all_credentials = Array.wrap(credentials).each_with_object({}) do |creds, hash|
-          auth_type = creds.delete(AUTH_TYPE_ATTR) || DEFAULT_AUTH_TYPE
-          creds.symbolize_keys!
-          creds.reverse_merge!(:userid => host.authentication_userid(auth_type))
-          hash[auth_type.to_sym] = creds
+        if authentications.present?
+          authentications.deep_symbolize_keys!
+          host.update_authentication(authentications)
         end
-        # end legacy ui-classic. if they provided the newer authentications, it will overwrite
-        all_credentials, _ = symbolize_password_keys!(authentications) if authentications
-        host.update_authentication(all_credentials) if all_credentials.present?
       end
     end
 
     def verify_credentials_resource(type, id = nil, data = {})
       api_resource(type, id, "Verifying Credentials for") do |host|
-        remember_host = data["remember_host"] == "true"
-        authentications, auth_type = symbolize_password_keys!(data[AUTH_ATTR])
-        {:task_id => host.verify_credentials_task(User.current_userid, auth_type, :credentials => authentications, :remember_host => remember_host)}
+        auth_type = data["authentications"].keys.first
```

Comment:
NOTE In the future I'd like to make `Host#verify_credentials` as close as possible to `ExtManagementSystem#verify_credentials` which takes just the data payload and no auth_type.